### PR TITLE
🧪 adds dagster-duckdb-polars library

### PR DESCRIPTION
**Pull request by pzarabadip**
### Summary & Motivation
For the performance reasons, we use `Polars` instead of `Pandas`. 
This PR provides a library as `dagster-duckdb-polars` which is built on top of the existing `dagster-duckdb-pandas` library to enable support of `Polars DataFrame` in `Dagster.

As a quick side fix for `dagster-duckdb-pandas`, I also replaced the deprecated `itertimes` in pandas with the `items`. 

### How I Tested These Changes
Tests for both libraries passed on my local machine and codes are formatted by running `make black`.

### CODE REVIEW

**Comment by jamiedemaria** [comment]
awesome! I'll take a look at this tomorrow

File: python_modules/libraries/dagster-duckdb-polars/MANIFEST.in [hunk 1]
```diff
@@ -0,0 +1,2 @@
+include LICENSE
+include dagster_duckdb_pandas/py.typed
```
Comment (jamiedemaria) [inline python_modules/libraries/dagster-duckdb-polars/MANIFEST.in:2]:
```suggestion
include dagster_duckdb_polars/py.typed
```

**Review by jamiedemaria — COMMENTED** [review]
This looks great! I'm going to push a commit to add the api docs to our docs site and then run CI. if that all passes then you'll be good to go

**Comment by pzarabadip** [comment]
Thanks very much @jamiedemaria.

**Comment by jamiedemaria** [comment]
hey @pzarabadip just wanted to give you an update on this! I merged in some changes to the DB IO manager parent class late last week that allow for static and multi partition support. I'm going to add a couple tests to this PR to add some test coverage for those new cases. I'll need to bring in another dagster engineer to do final review and then this will be good to go in

**Comment by pzarabadip** [comment]
Hi @jamiedemaria. Thanks for the update.

File: python_modules/libraries/dagster-duckdb-pandas/dagster_duckdb_pandas/duckdb_pandas_type_handler.py [hunk 2]
```diff
@@ -48,8 +48,8 @@ def handle_output(self, context: OutputContext, table_slice: TableSlice, obj: pd
                 "dataframe_columns": MetadataValue.table_schema(
                     TableSchema(
                         columns=[
-                            TableColumn(name=name, type=str(dtype))
-                            for name, dtype in obj.dtypes.iteritems()
+                            TableColumn(name=str(name), type=str(dtype))
```
Comment (benpankow) [inline python_modules/libraries/dagster-duckdb-pandas/dagster_duckdb_pandas/duckdb_pandas_type_handler.py:51]: is this change intentional?

**Review by benpankow — APPROVED** [review]
great!

File: python_modules/libraries/dagster-duckdb-pandas/dagster_duckdb_pandas/duckdb_pandas_type_handler.py (hunk 2 above)
Comment (pzarabadip) [reply python_modules/libraries/dagster-duckdb-pandas/dagster_duckdb_pandas/duckdb_pandas_type_handler.py:51]: Thanks @benpankow. `itertimes` is deprecated since pandas version 1.5.0, hence I changed it to `items`.
However, I'm thinking now that it might result in issue for users with an older version of pandas. I can revert it for now but keeping a note to change it in future?

Comment (jamiedemaria) [reply python_modules/libraries/dagster-duckdb-pandas/dagster_duckdb_pandas/duckdb_pandas_type_handler.py:51]: hey @pzarabadip i'm going to go ahead and revert this change so we can merge in the PR - I'll make a note of it so i can address it in a follow up pr for cleaning up this integration

Comment (pzarabadip) [reply python_modules/libraries/dagster-duckdb-pandas/dagster_duckdb_pandas/duckdb_pandas_type_handler.py]: hey @jamiedemaria Sure and thanks.
